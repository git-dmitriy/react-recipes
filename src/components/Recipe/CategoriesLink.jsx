import { Link } from 'react-router-dom';

export const CategoriesLink = ({ country, category }) => {
  return (
    <div>
      <p>
        <span className='inline-block w-19'>Country:</span>{' '}
        <span className='font-bold'>
          {country === 'Unknown' ? (
            'Origin not establish'
          ) : (
            <Link
              to={`/country/${country}`}
              className='border-b border-current hover:opacity-50 transition-opacity'
            >
              {country}
            </Link>
          )}
        </span>
      </p>
      <p>
        <span className='inline-block w-19'>Category:</span>{' '}
        <span className='font-bold'>
          {
            <Link
              to={`/category/${category}`}
              className='border-b border-current hover:opacity-50 transition-opacity'
            >
              {category}
            </Link>
          }
        </span>
      </p>
    </div>
  );
};
