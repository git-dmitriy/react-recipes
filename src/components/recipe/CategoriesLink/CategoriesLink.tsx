import { Link } from 'react-router-dom';
import { CategoryLinkTypes } from 'appTypes';

export const CategoriesLink: React.FC<CategoryLinkTypes> = ({
  country,
  category,
}) => {
  return (
    <div>
      <p>
        <span className='inline-block w-19'>Country:</span>{' '}
        <span className='font-bold'>
          {country === 'Unknown' ? (
            'Origin not establish'
          ) : (
            <Link
              to={`/country/${country}`}
              className='border-b border-current hover:opacity-50 transition-opacity'
            >
              {country}
            </Link>
          )}
        </span>
      </p>
      <p>
        <span className='inline-block w-19'>Category:</span>{' '}
        <span className='font-bold'>
          {
            <Link
              to={`/category/${category}`}
              className='border-b border-current hover:opacity-50 transition-opacity'
            >
              {category}
            </Link>
          }
        </span>
      </p>
    </div>
  );
};
