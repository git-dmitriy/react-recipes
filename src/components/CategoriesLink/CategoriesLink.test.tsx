import renderer from 'react-test-renderer';
import {MemoryRouter} from 'react-router-dom';
import {CategoriesLink} from './CategoriesLink';
import {it, expect} from 'vitest';

const country = 'British';
const category = 'Breakfast';

it('should render category links', () => {
    const tree = renderer
        .create(
            <MemoryRouter>
                <CategoriesLink country={country} category={category}/>
            </MemoryRouter>
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});

it('should render category links without country origin', () => {
    const country = 'Unknown';

    const tree = renderer
        .create(
            <MemoryRouter>
                <CategoriesLink country={country} category={category}/>
            </MemoryRouter>
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});
