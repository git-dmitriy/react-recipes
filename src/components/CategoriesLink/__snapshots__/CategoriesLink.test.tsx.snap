// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should render category links 1`] = `
<div>
  <p>
    <span
      className="inline-block w-19"
    >
      Country:
    </span>
     
    <span
      className="font-bold"
    >
      <a
        className="border-b border-current hover:opacity-50 transition-opacity"
        href="/country/British"
        onClick={[Function]}
      >
        British
      </a>
    </span>
  </p>
  <p>
    <span
      className="inline-block w-19"
    >
      Category:
    </span>
     
    <span
      className="font-bold"
    >
      <a
        className="border-b border-current hover:opacity-50 transition-opacity"
        href="/category/Breakfast"
        onClick={[Function]}
      >
        Breakfast
      </a>
    </span>
  </p>
</div>
`;

exports[`should render category links without country origin 1`] = `
<div>
  <p>
    <span
      className="inline-block w-19"
    >
      Country:
    </span>
     
    <span
      className="font-bold"
    >
      Origin not establish
    </span>
  </p>
  <p>
    <span
      className="inline-block w-19"
    >
      Category:
    </span>
     
    <span
      className="font-bold"
    >
      <a
        className="border-b border-current hover:opacity-50 transition-opacity"
        href="/category/Breakfast"
        onClick={[Function]}
      >
        Breakfast
      </a>
    </span>
  </p>
</div>
`;
